src/test/java/io/github/perrymant/moneymaker/ApplicationTest.java

```diff
@@ -55,10 +66,16 @@ public void doesntPrintHelpWithNoArgs() {
         Assert.assertNotEquals(HELP_MESSAGE, logger.getMessage());
     }
 
+    @Test
+    public void printsReportIfReportIsGiven() {
+        target.start(new String[]{"report"});
+        Assert.assertEquals(REPORT, logger.getMessage());
```

Comment:
Doesn't this check depend upon how much data the system currently has? Might have to use the dependency injection trick to use a custom `Budget` class as well.